repository/src/main/java/com/outdoor/foodcalc/domain/repository/plan/FoodPlanRepo.java

```diff
@@ -1,46 +1,127 @@
-        return Optional.empty();
+        SqlParameterSource parameters = new MapSqlParameterSource().addValue("id", id);
+        return getSingleObject(SELECT_FOOD_PLAN_SQL, parameters, this);
     }
 
     @Override
     public long addFoodPlan(FoodPlan plan) {
-        return 0;
+        KeyHolder holder = getKeyHolder();
+        MapSqlParameterSource parameters = getSqlParameterSource(plan);
+        jdbcTemplate.update(INSERT_FOOD_PLAN_SQL, parameters, holder, new String[] {"id"});
+        Number key = holder.getKey();
+        return key != null ? key.longValue() : -1L;
+    }
+
+    MapSqlParameterSource getSqlParameterSource(FoodPlan plan) {
+        return new MapSqlParameterSource()
+                .addValue("id", plan.getId())
+                .addValue("name", plan.getName())
+                .addValue("members", plan.getMembers())
+                .addValue("createdOn", plan.getCreatedOn().toOffsetDateTime())
+                .addValue("lastUpdated", plan.getLastUpdated().toOffsetDateTime())
+                .addValue("description", plan.getDescription());
     }
 
     @Override
-    public boolean updateFoodPlan(FoodPlan foodPlan) {
-        return false;
+    public boolean updateFoodPlan(FoodPlan plan) {
+        MapSqlParameterSource parameters = new MapSqlParameterSource()
+                .addValue("id", plan.getId())
+                .addValue("name", plan.getName())
+                .addValue("members", plan.getMembers())
+                .addValue("lastUpdated", plan.getLastUpdated().toOffsetDateTime())
+                .addValue("description", plan.getDescription());
+        return jdbcTemplate.update(UPDATE_FOOD_PLAN_SQL, parameters) > 0;
     }
 
     @Override
     public boolean deleteFoodPlan(long id) {
-        return false;
+        SqlParameterSource parameters = new MapSqlParameterSource().addValue("id", id);
+        return jdbcTemplate.update(DELETE_FOOD_PLAN_SQL, parameters) > 0;
     }
 
     @Override
     public boolean existsFoodPlan(long id) {
-        return false;
+        SqlParameterSource parameters = new MapSqlParameterSource().addValue("id", id);
+        Long count = jdbcTemplate.queryForObject(SELECT_FOOD_PLAN_EXIST_SQL, parameters, Long.class);
+        return count != null && count > 0;
     }
 
     @Override
     public boolean saveLastUpdated(long planId, ZonedDateTime lastUpdated) {
-        return false;
+        MapSqlParameterSource parameters = new MapSqlParameterSource()
+                .addValue("id", planId)
+                .addValue("lastUpdated", lastUpdated.toOffsetDateTime());
+        return jdbcTemplate.update(UPDATE_FOOD_PLAN_LAST_UPDATED_SQL, parameters) > 0;
+    }
+
+    @Override
+    public FoodPlan mapRow(ResultSet resultSet, int rowNum) throws SQLException {
+        OffsetDateTime odtCreatedOn = resultSet.getObject("createdOn", OffsetDateTime.class);
+        ZonedDateTime createdOn = odtCreatedOn.toZonedDateTime();
```

Comment:
please inline this variable in builder:
`.createdOn(odtCreatedOn.toZonedDateTime())`

Reply:
and the same for updated